python-3.10/Dockerfile

```diff
@@ -7,42 +7,40 @@ RUN curl https://packages.microsoft.com/keys/microsoft.asc | apt-key add - \
     && curl https://packages.microsoft.com/config/debian/10/prod.list > /etc/apt/sources.list.d/mssql-release.list \
     && apt-get update && ACCEPT_EULA=Y apt-get install -y --no-install-recommends \
         libgeos-c1v5 \
-        python-numpy \
-        python-scipy \
-        python-matplotlib \
-        ipython \
         msodbcsql17 \
         mssql-tools \
-        python-pandas \
-        python-sympy \
-        python-nose \
         g++ \
         libsasl2-dev \
         libatlas-base-dev \
         libicu-dev \
         gfortran \
         unixodbc-dev \
-        python3-icu \
         wget \
         git \
     && rm -rf /var/lib/apt/lists/*
 
 ENV PATH $PATH:/opt/mssql-tools/bin
-# https://pythonspeed.com/articles/activate-virtualenv-dockerfile/
-ENV VIRTUAL_ENV=/home/default
-ENV PATH="$VIRTUAL_ENV/bin:$PATH"
+
+ENV VIRTUAL_ENV /home/default
+
+# prep venv dir
+RUN mkdir $VIRTUAL_ENV \
+    && chown :users -R $VIRTUAL_ENV \
+    && chmod a+rwx -R $VIRTUAL_ENV
 
 # From https://jdk.java.net/13/, stolen from https://github.com/docker-library/openjdk/blob/master/8/jdk/Dockerfile#L22
 ENV JAVA_HOME /usr/local/openjdk
+
 ENV PATH $JAVA_HOME/bin:$PATH
+
 RUN wget https://download.java.net/java/GA/jdk13.0.2/d4173c853231432d94f001e99d882ca7/8/GPL/openjdk-13.0.2_linux-x64_bin.tar.gz \
     && mkdir $JAVA_HOME \
     && tar xv --file openjdk-13*_bin.tar.gz --directory "$JAVA_HOME" --no-same-owner --strip-components 1 \
     && find "$JAVA_HOME/lib" -name '*.so' -exec dirname '{}' ';' | sort -u > /etc/ld.so.conf.d/docker-openjdk.conf \
     && ldconfig
 
-RUN mkdir /home/default \
-	&& python3 -m venv $VIRTUAL_ENV \
+RUN python3 -m venv $VIRTUAL_ENV \
```

Comment:
What if we remove above
```
# prep venv dir
RUN mkdir $VIRTUAL_ENV \
    && chown :users -R $VIRTUAL_ENV \
    && chmod a+rwx -R $VIRTUAL_ENV
```

And put the mkdir here
```
RUN mkdir $VIRTUAL_ENV \
 && python3 -m venv $VIRTUAL_ENV \
 && . $VIRTUAL_ENV/bin/activate \
...
```

You run as `root` here, so I see no point in `chmod` & `chown`. It's again at the end of this `RUN`

Reply:
I see it's done that way in 3.8 :slightly_smiling_face: